rust/opendp/src/meas/histogram_alp/mod.rs

```diff
@@ -0,0 +1,398 @@
+          T: 'static + num::Float + DistanceConstant<T> + CastInternalReal + InfCast<C>,
+          AlpState<K,T> : CheckNull {
+    
+    if alpha.is_sign_negative() || alpha.is_zero() {
+        return fallible!(MakeMeasurement, "alpha must be positive")
+    }
+    if scale.is_sign_negative() || scale.is_zero() {
+        return fallible!(MakeMeasurement, "scale must be positive")
+    }
+    if s == 0 {
+        return fallible!(MakeMeasurement, "s can not be zero")
+    }
+    if check_parameters(alpha, scale) {
+        return fallible!(MakeMeasurement, "scale divided by alpha must be above 2^-52")
+    }
+    
+    Ok(Measurement::new(
+        SizedDomain::new(MapDomain { key_domain: AllDomain::new(), value_domain: AllDomain::new() }, n),
+        AllDomain::new(),
+        Function::new_fallible(move |x: &HashMap<K, C>| {
+            let z = compute_projection(x, &h, alpha, scale, s)?;
+            Ok(AlpState { alpha, scale, h:h.clone(), z })
+        }),
+        L1Distance::default(),
+        MaxDivergence::default(),
+        PrivacyRelation::new_from_constant(scale)
+    ))
+}
+
+pub fn make_alp_histogram_parameterized<K, C, T>(n: usize, alpha: T, scale: T, beta: C, size_factor: u32) 
+        -> Fallible<Measurement<SizedHistogramDomain<K, C>, 
+                                AlpDomain<K, T>, 
+                                L1Distance<C>, MaxDivergence<T>>>
+    where K: 'static + Eq + Hash + Clone + CheckNull,
+          C: 'static + Clone + Integer + CheckNull + DistanceConstant<C> + InfCast<T> + ToPrimitive,
+          T: 'static + num::Float + DistanceConstant<T> + CastInternalReal + InfCast<C>,
+          AlpState<K,T> : CheckNull {
+
+    let beta: f64 = T::inf_cast(beta).unwrap().to_f64()
+        .ok_or_else(|| err!(MakeTransformation, "failed to parse beta"))?;
+    let quotient = (scale / alpha).to_f64()
+        .ok_or_else(|| err!(MakeTransformation, "failed to parse scale/alpha"))?;
+    let m = (beta * quotient).ceil() as usize;
+    
+    let exp = exponent_next_power_of_two((size_factor as f64 * n as f64 * quotient) as u64);
+    let h = (0..m).map(|_| sample_hash_function(exp)).collect::<Fallible<HashFunctions<K>>>()?;
+
+    make_alp_histogram(n, alpha, scale, 1 << exp, h)
+}
+
+pub fn make_alp_histogram_simple<K, C, T>(n: usize, scale: T, beta: C) 
+        -> Fallible<Measurement<SizedHistogramDomain<K, C>, 
+                                AlpDomain<K, T>, 
+                                L1Distance<C>, MaxDivergence<T>>>
+    where K: 'static + Eq + Hash + Clone + CheckNull,
+          C: 'static + Clone + Integer + CheckNull + DistanceConstant<C> + InfCast<T> + ToPrimitive,
+          T: 'static + num::Float + DistanceConstant<T> + CastInternalReal + InfCast<C>,
+          AlpState<K,T> : CheckNull {
+    
+    make_alp_histogram_parameterized(n, T::from(ALPHA_DEFAULT).unwrap(), scale, beta, SIZE_FACTOR_DEFAULT)
+}
```

Comment:
You could omit this entire constructor if `alpha` and `size_factor` in `make_alp_histogram_parameterized` were adjusted option types:

```rust
let alpha = if let Some(a) = alpha {a} else {T::inf_cast(ALPHA_DEFAULT)?)};
let size_factor = size_factor.unwrap_or(SIZE_FACTOR_DEFAULT);
```

Reply:
Transferred from DM's: 
I have some thoughts about naming!
- `make_alp_histogram_simple` could be folded into `make_alp_histogram_parameterized` by making two of the args optional
- I feel like most people will use the parameterized constructor, so what do you think about renaming as follows:
    * `make_alp_histogram` -> `make_alp_histogram_with_hashers`
    * `make_alp_histogram_parameterized` -> `make_alp_histogram`
- Does this implementation work with any integer query? It seems like it would work with, for example, grouped integer sums
    * Do you think `make_base_alp*` would be a better prefix? Or is that too general?

So a typical analyst flow would be `make_count_by(...) >> make_base_alp(...)` or `make_sum_by(...) >> make_base_alp(...)` and if they also wanted to specify hashers, then they could run `make_base_alp_with_hashers(...)`.